src/commands/up.rs

```diff
@@ -117,22 +117,49 @@ pub async fn command(args: Args, _json: bool) -> Result<()> {
         println!("Indexing...");
         None
     };
+
+    // Explanation for the below block
+    // arc is a reference counted pointer to a mutexed vector of bytes, which
+    // stores the actual tarball in memory.
+    //
+    // parz is a parallelized gzip writer, which writes to the arc (still in memory)
+    //
+    // archive is a tar archive builder, which writes to the parz writer `new(&mut parz)
+    //
+    // builder is a directory walker which returns an iterable that we loop over to add
+    // files to the tarball (archive)
+    //
+    // during the iteration of `builder`, we ignore all files that match the patterns found in
+    // .railwayignore
+    // .gitignore
+    // .git/**
+    // node_modules/**
     let bytes = Vec::<u8>::new();
     let arc = Arc::new(Mutex::new(bytes));
     let mut parz = ZBuilder::<Gzip, _>::new()
         .num_threads(num_cpus::get())
         .from_writer(SynchronizedWriter::new(arc.clone()));
+
+    // list of all paths to ignore by default
+    let ignore_paths = vec![".git", "node_modules"];
+    let ignore_paths: Vec<&std::ffi::OsStr> = ignore_paths
+        .iter()
+        .map(|s| std::ffi::OsStr::new(s))
+        .collect();
+
     {
         let mut archive = Builder::new(&mut parz);
         let mut builder = WalkBuilder::new(path);
         builder.add_custom_ignore_filename(".railwayignore");
         builder.add_custom_ignore_filename(".gitignore");
+
         let walker = builder.follow_links(true).hidden(false);
         let walked = walker.build().collect::<Vec<_>>();
         if let Some(spinner) = spinner {
             spinner.finish_with_message("Indexed");
         }
         if std::io::stdout().is_terminal() {
+            //#region
```

Comment:
What is this for?

Reply:
An extension I was using to collapse the #region. I'll remove it on a quick push during lunch. Sorry